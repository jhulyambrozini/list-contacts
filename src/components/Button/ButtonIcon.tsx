type ButtonIconProps = {
  icon: string
  alt: string
}

const ButtonIcon = ({ icon, alt }: ButtonIconProps) => (
  <img src={icon} alt={alt} />
)

export default ButtonIcon
