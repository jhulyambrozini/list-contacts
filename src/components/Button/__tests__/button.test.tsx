import '@testing-library/jest-dom'
import {
  cleanup,
  fireEvent,
  getByRole,
  getByText,
  render,
  screen
} from '@testing-library/react'
import { Button, ButtonProps } from '..'
import ButtonIcon from '../ButtonIcon'
import { vi } from 'vitest'
import userEvent from '@testing-library/user-event'

describe('<Button />', () => {
  afterEach(() => cleanup())
  it('deve renderizar primary buttton com label corretamente', () => {
    render(
      <Button.Primary padding="2rem" type="button" title="Click me">
        <Button.Label label="Primary" />
      </Button.Primary>
    )

    expect(screen.getByText(/primary/i)).toBeInTheDocument()
  })

  it('deve renderizar primary buttton com label e icon corretamente', () => {
    render(
      <Button.Primary padding="2rem" type="button" title="Click me">
        <Button.Label label="Primary with label and icon" />
        <ButtonIcon icon="/img.png" alt="icon" />
      </Button.Primary>
    )

    expect(screen.getByText(/primary with label and icon/i)).toBeInTheDocument()
    expect(screen.getByRole('img', { name: /icon/i })).toBeInTheDocument()
  })

  it('deve renderizar seconadry buttton com label corretamente', () => {
    render(
      <Button.Secondary padding="2rem" type="button" title="Click me">
        <Button.Label label="Secondary with label" />
      </Button.Secondary>
    )

    expect(screen.getByText(/secondary/i)).toBeInTheDocument()
  })

  it('deve renderizar Secondary buttton com icon corretamente', () => {
    render(
      <Button.Secondary padding="2rem" type="button" title="Click me">
        <ButtonIcon icon="/img.png" alt="icon" />
      </Button.Secondary>
    )

    expect(screen.getByRole('img', { name: /icon/i })).toBeInTheDocument()
  })

  it('deve renderizar circle button com icon corretamente', () => {
    render(
      <Button.Circle padding="2rem" type="button" title="Click me">
        <ButtonIcon icon="/img.png" alt="icon" />
      </Button.Circle>
    )

    expect(screen.getByRole('img', { name: /icon/i })).toBeInTheDocument()
  })

  it('deve renderizar circle button com label corretamente', () => {
    render(
      <Button.Circle padding="2rem" type="button" title="Click me">
        <Button.Label label="Circle with label" />
      </Button.Circle>
    )

    expect(screen.getByText(/circle/i)).toBeInTheDocument()
  })

  it('should calls onClick handler when button is clicked', () => {
    const handleClick = vi.fn()
    render(
      <Button.Primary
        padding="2rem"
        type="button"
        title="Click me"
        onclick={handleClick}
      >
        <Button.Label label="button" />
      </Button.Primary>
    )

    fireEvent.click(screen.getByText('button'))
    expect(handleClick).toHaveBeenCalled()
  })
})
