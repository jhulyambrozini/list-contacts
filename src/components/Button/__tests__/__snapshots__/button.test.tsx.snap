// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Button /> > should render Secondary button with icon correctly 1`] = `
<button
  class="sc-bcPKhP sc-dkjKgF gxyhQS jtVglq"
  padding="2rem"
  title="Click me"
  type="button"
>
  <img
    alt="icon"
    src="/img.png"
  />
</button>
`;

exports[`<Button /> > should render circle button with icon correctly 1`] = `
<button
  class="sc-bcPKhP sc-hLclGa gxyhQS bSMmYf"
  padding="2rem"
  title="Click me"
  type="button"
>
  <img
    alt="icon"
    src="/img.png"
  />
</button>
`;

exports[`<Button /> > should render circle button with label correctly 1`] = `
<button
  class="sc-bcPKhP sc-hLclGa gxyhQS bSMmYf"
  padding="2rem"
  title="Click me"
  type="button"
>
  <span>
    Circle with label
  </span>
</button>
`;

exports[`<Button /> > should render primary button with label and icon correctly 1`] = `
<button
  class="sc-bcPKhP sc-grXZZQ gxyhQS eCXYuZ"
  padding="2rem"
  title="Click me"
  type="button"
>
  <span>
    Primary with label and icon
  </span>
  <img
    alt="icon"
    src="/img.png"
  />
</button>
`;

exports[`<Button /> > should render primary buttton with label correctly 1`] = `
<button
  class="sc-bcPKhP sc-grXZZQ gxyhQS eCXYuZ"
  padding="2rem"
  title="Click me"
  type="button"
>
  <span>
    Primary
  </span>
</button>
`;

exports[`<Button /> > should render secondary button with label correctly 1`] = `
<button
  class="sc-bcPKhP sc-dkjKgF gxyhQS jtVglq"
  padding="2rem"
  title="Click me"
  type="button"
>
  <span>
    Secondary with label
  </span>
</button>
`;
